controller/src/controller.rs

```diff
@@ -104,16 +106,16 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
     /// The state transition is then attempted. If successful, this node should be detected as part of the active
     /// set during the next iteration of the controller's event loop.
     #[instrument(skip(self))]
-    async fn find_and_update_ready_node(&self) -> Option<BottlerocketShadow> {
-        let mut shadows: Vec<BottlerocketShadow> = self.all_nodes();
+    async fn find_and_update_ready_node(&self) -> Result<Option<BottlerocketShadow>> {
+        let shadows: Vec<BottlerocketShadow> = self.all_nodes();
 
-        shadows.sort_by(|a, b| a.compare_crash_count(b));
-        for brs in shadows {
+        sort_shadows(shadows.clone(), &get_associated_bottlerocketshadow_name()?)?;
```

Comment:
This won't actually sort the vector, since we are cloning it as input rather than using a mutable reference.

Reply:
Yeah, I remember I changed it to `let sorted_shadows = sort_shadows(shadows.clone(), &get_associated_bottlerocketshadow_name()?)?;`. Maybe I was doing wrong with rebase. Thanks I'll fix it.

Reply:
I think it would be more idiomatic here to use a mutable reference as input to the function. You can just return an empty `Result<()>` to indicate whether something or not went wrong.